src/actinia_core/resources/ephemeral_processing.py

```diff
@@ -536,42 +534,9 @@ def _send_to_database(self, document, final=False):
         # Call the webhook after the final result was send to the database
         try:
             if final is True and self.webhook_finished is not None:
-                self.message_logger.info(
-                    "Send POST request to finished webhook url: %s"
-                    % self.webhook_finished)
-                http_code, response_model = pickle.loads(document)
-                if self.webhook_auth:
-                    # username is expected to be without colon (':')
-                    r = requests.post(
-                        self.webhook_finished, json=json.dumps(response_model),
-                        auth=HTTPBasicAuth(
-                            self.webhook_auth.split(':')[0],
-                            ':'.join(self.webhook_auth.split(':')[1:])))
-                else:
-                    r = requests.post(self.webhook_finished,
-                                      json=json.dumps(response_model))
-                if r.status_code not in [200, 204]:
-                    raise AsyncProcessError(
-                        "Unable to access finished webhook URL %s"
-                        % self.webhook_finished)
+                self._post_to_webhook(document, 'finished')
             elif final is False and self.webhook_update is not None:
-                self.message_logger.info(
-                    "Send POST request to update webhook url: %s" % self.webhook_update)
-                http_code, response_model = pickle.loads(document)
-                if self.webhook_auth:
-                    # username is expected to be without colon (':')
-                    r = requests.post(
-                        self.webhook_update, json=json.dumps(response_model),
-                        auth=HTTPBasicAuth(
-                            self.webhook_auth.split(':')[0],
-                            ':'.join(self.webhook_auth.split(':')[1:])))
-                else:
-                    r = requests.post(self.webhook_update,
-                                      json=json.dumps(response_model))
-                if r.status_code not in [200, 204]:
-                    raise AsyncProcessError(
-                        "Unable to access the update webhook URL %s"
-                        % self.webhook_update)
+                self._post_to_webhook(document, 'update')
```

Comment:
Great first split up! This makes totally sense. Hopefully we can discover more like these.